astropy/units/quantity_helper/converters.py

```diff
@@ -58,7 +58,7 @@ def import_module(self, module):
             Name of the module. Has to have been registered beforehand.
         """
         with self._lock:
-            module_info = self.modules.pop(module)
+            module_info = self.modules[module]
```

Comment:
I'm super confused why this would be necessary. What test fails if you keep it? It should not be possible to get here more than once...

Reply:
`test_scipy_registered` is the test that fails without this. See the reprod invoke in the PR description.

Reply:
And yes, we don't get here more than once, but that test explicitly checks for a state that is changed by this line.

Reply:
This makes sense too, but I don't get what you don't like about my less destructive `import_module` ?

Reply:
The advantage of what I have here is that, if ever something when wrong with checking membership, we will know, rather than everything becoming very slow as for every call the whole module gets loaded back in.

But, more generally, I prefer not to change code to fix the double tests if it is arguably the test that was written sloppily. We cannot know whether somewhere in the wild code relies on the current behaviour.